src/sentry/incidents/models/alert_rule.py

```diff
@@ -496,6 +496,59 @@ def get_queryset(self) -> BaseQuerySet[AlertRuleTriggerAction]:
         return super().get_queryset().exclude(status=ObjectStatus.PENDING_DELETION)
 
 
+class AlertRuleActionHandlerFactory(abc.ABC):
+    """A factory for action handlers tied to a specific service.
+
+    The factory's builder method is augmented with metadata about which service it is
+    for and which target types that service supports.
+    """
+
+    def __init__(
+        self,
+        slug: str,
+        service_type: ActionService,
+        supported_target_types: Iterable[ActionTarget],
+        integration_provider: str | None,
+    ) -> None:
+        self.slug = slug
+        self.service_type = service_type
+        self.supported_target_types = frozenset(supported_target_types)
+        self.integration_provider = integration_provider
+
+    @abc.abstractmethod
+    def build_handler(
+        self,
+        action: AlertRuleTriggerAction,
+        incident: Incident,
+        project: Project,
+    ) -> ActionHandler:
+        raise NotImplementedError
+
+
+class _AlertRuleActionHandlerClassFactory(AlertRuleActionHandlerFactory):
```

Comment:
The above name might make sense, since now this is a factory specific to alert rules?